frame/support/src/dispatch.rs

```diff
@@ -1265,28 +1265,30 @@ macro_rules! decl_module {
 	};
 
 	(@impl_on_initialize
+		{ $system:ident }
 		$module:ident<$trait_instance:ident: $trait_name:ident$(<I>, $instance:ident: $instantiable:path)?>;
 		{ $( $other_where_bounds:tt )* }
 		fn on_initialize() -> $return:ty { $( $impl:tt )* }
 	) => {
-		impl<$trait_instance: $trait_name$(<I>, $instance: $instantiable)?>
-			$crate::traits::OnInitialize<$trait_instance::BlockNumber>
+		impl<$trait_instance: $trait_name$(<I>, $instance: $instantiable)? + $system::Trait>
```

Comment:
this bound doesn't work for instantiable module it basically expand to `Trait<I>, I: Instance + system::Trait`
```suggestion
		impl<$trait_instance: $system::Trait + $trait_name$(<I>, $instance: $instantiable)?>
```

Reply:
@thiolliere why do we use `$instance` in one place but `<I>` in another? Should be:

```
(<$instance>, $instance: $instantiable)?

Reply:
indeed it should be done as you say. Currently we don't support other instance name than `I`.